src/tools/interop/idt/discovery/matter_mdns.py

```diff
@@ -0,0 +1,86 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from zeroconf import ServiceBrowser, ServiceInfo, ServiceListener, Zeroconf
+
+_MDNS_TYPES = {
+    "_matterd._udp.local.": "COMMISSIONER",
+    "_matterc._udp.local.": "COMMISSIONABLE",
+    "_matter._tcp.local.": "OPERATIONAL",
+    "_meshcop._udp.local.": "THREAD_BORDER_ROUTER",
+}
+
+
+class MatterMdnsListener(ServiceListener):
+
+    def __init__(self, artifact_dir: str) -> None:
+        super().__init__()
+        self.artifact_dir = artifact_dir
+        self.logger = logging.getLogger(__file__)
+
+    @staticmethod
+    def log_addr(info: ServiceInfo) -> str:
+        ret = "\n"
+        for addr in info.parsed_scoped_addresses():
+            ret += f"{addr}\n"
+        return ret
+
+    @staticmethod
+    def log_vid_pid(info: ServiceInfo) -> str:
```

Comment:
Does the rest of the TXT packet get logged? It would be helpful

Reply:
Yes I'm logging everything, although not parsing every field.

E.x.

```
Browsing Matter mDNS, press enter to stop
2023-09-18 14:09:42,233.233 INFO {matter_mdns} [handle_service_info]
Google Nest Hub Max (a6f7)._meshcop._udp.local.
b'vcd':	b'106A6D1B9B99FEAA301026F7D2DDE8CD'
b'id':	b'\x10jm\x1b\x9b\x99\xfe\xaa0\x10&\xf7\xd2\xdd\xe8\xcd'
b'vo':	b'|\xd9\\'
b'vvo':	b'7CD95C'
b'vxp':	b'1dc3269b3297a6d6'
b'vat':	b'000064cd6e100d51'
b'rv':	b'1'
b'tv':	b'1.3.0'
b'sb':	b'\x00\x00\x001'
b'nn':	b'NEST-PAN-C15C'
b'xp':	b'\x1d\xc3&\x9b2\x97\xa6\xd6'
b'vn':	b'Google Inc.'
b'mn':	b'Google Nest Hub Max'
b'xa':	b'\x02\x08\xbd\xc7\x10N\xa6\xf7'
b'at':	b'\x00\x00d\xcdn\x10\rQ'
b'pt':	b'cfb\x98'
*************
SERVICE ADDED
THREAD_BORDER_ROUTER
192.168.8.186
fe80::365e:eb9:8e61:91ad
fd60:7413:2082:8:d275:d508:590:907
fd60:7413:2082:8:6158:e6ed:ce7c:d37b
fd1d:c326:9b32:a6d6:5ddb:e795:69b:8df9
fd1d:c326:9b32:a6d6:39d0:4a9f:1eab:eabb
 

2023-09-18 14:09:42,234.234 INFO {matter_mdns} [handle_service_info]
61701bf6702881af-000000001a3fbb61._matter._tcp.local.
*************
SERVICE ADDED
OPERATIONAL
192.168.8.186
fe80::365e:eb9:8e61:91ad
fd60:7413:2082:8:d275:d508:590:907
fd60:7413:2082:8:6158:e6ed:ce7c:d37b
fd1d:c326:9b32:a6d6:5ddb:e795:69b:8df9
fd1d:c326:9b32:a6d6:39d0:4a9f:1eab:eabb
```